src/vision/calib/Calibration.cpp

```diff
@@ -0,0 +1,210 @@
+#include "MonoCalibrator.hpp"
+#include "wfov_camera_msgs/WFOVImage.h"
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <string>
+#include <vector>
+
+using namespace cv_bridge;
+
+Settings s;
+
+bool calibFinished = false;
+vector<vector<Point2f>> imagePoints;
+Mat cameraMatrix, distCoeffs;
+Size imageSize;
+int mode = CAPTURING;
+clock_t prevTimestamp = 0;
+const Scalar RED(0, 0, 255), GREEN(0, 255, 0);
+const char ESC_KEY = 27;
+
+void imageCallback(const wfov_camera_msgs::WFOVImage::ConstPtr& msg)
+{
+    ROS_DEBUG_STREAM("Callback, Mode: " << mode);
+    Mat view = toCvShare(msg->image, msg,
+                         sensor_msgs::image_encodings::BGR8)->image;
+
+    if (mode == CAPTURING && imagePoints.size() >= (size_t)s.nrFrames)
+    {
+        ROS_DEBUG_STREAM("Calibrating and saving during capture");
+        if (runCalibrationAndSave(s, imageSize, cameraMatrix, distCoeffs,
+                                  imagePoints))
+            mode = CALIBRATED;
+        else
+            mode = DETECTION;
+    }
```

Comment:
@ryan-summers:  you are the one with the weird spacing.  https://google.github.io/styleguide/cppguide.html#Function_Calls